### ISSUE TITLE
Should not fail when there are no classes for inspection

### ISSUE THREAD

**Issue by boris-petrov** [spotbugs/spotbugs#525]
Whenever I try running SpotBugs on a sourceset that has no (Java) classes, it fails with:

```
No classes configured for SpotBugs analysis.
```

I don't see a reason for that to be an error. This could be silently ignored. I do something like:

```groovy
def tasks = sourceSets
	.collect { it.name.capitalize() }
	.collect { ['pmd' + it, 'spotbugs' + it, 'checkstyle' + it] }
	.flatten()
```

But the SpotBugs task fails for some sourcesets so now I will have to write some checks and ignore such sourcesets only for SpotBugs...

**Comment by iloveeclipse** [comment]
One reason (I believe) for current behavior is to warn users about configuration issues. Without warning one might think "everything is OK" without looking further if any code was analyzed at all.

**Comment by boris-petrov** [comment]
Yes, I thought about that but all the rest of the tools (PMD, Checkstyle, CodeNarc, etc) don't do that so I guess you could follow in their steps. I do understand the reasoning though so I won't be mad if you decide you won't do it. :) Just some food for thought.

**Comment by boris-petrov** [comment]
Actually you already return a `NO-SOURCE` when SpotBugs is run on a sourceset that has only Groovy code. I'm a bit confused as to when SpotBugs blows up and when it prints `NO-SOURCE`...

**Comment by semen-levin-epam** [comment]
I'm suffering from the same issue. SpotBugs Gradle plugin ver 1.6.1. I have Multi-Project Gradle project and for some projects I see `NO-SOURCE`, but the same error for others. Please just do nothing if `classes` is an empty FileCollection.

**Comment by ThrawnCA** [comment]
Actually, Surefire will indeed complain if it has no tests to execute. So there's precedent.
